### ISSUE TITLE
Latest miniconda incompatible with mamba?

### ISSUE THREAD

**Issue by gernophil** [conda/conda#13041]
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

Hey everyone,
I've found something about this, but not really a duplicate or someone pointing out that there seems to be an incompatibility of the latest miniconda and mamba (from conda-forge). I know, you're not the mamba guys, but I think a lot of the people using conda actually also use mamba. Here's what happened:

I made a complete fresh install of the latest miniconda (23.5.2; x86_64) and installed `mamba` from `conda-forge`. I am unable to install anything using `mamba`, since it always throws this error:
```
ImportError: dlopen(~/miniconda3/lib/python3.11/site-packages/libmambapy/bindings.cpython-311-darwin.so, 2): Library not loaded: @rpath/libarchive.13.dylib Referenced from: ~/miniconda3/lib/libmamba.2.0.0.dylib Reason: image not found
```
On another machine I am running miniconda 23.1.0 in combination with mamba and it works perfect (and also did for a long time). So, I did the same on this machine and clean uninstalled miniconda 23.5.2 and replaced it with miniconda 23.1.0 and reinstalled mamba and now it works. There seems to be an incompatibility of mamba with the newest version of miniconda. Can anyone confirm this?

**Comment by mfansler** [comment]
It is odd, but in fact the case that even though Conda has embraced `libmamba` as its backend, the Mamba org no longer recommends using Miniconda **base** + Mamba (see [Mamba installation docs](https://mamba.readthedocs.io/en/latest/mamba-installation.html#existing-conda-install-not-recommended)). Instead, one should just go straight to Mambaforge (which is [now the default Miniforge installer](https://github.com/conda-forge/conda-forge.github.io/pull/1995)). The situation is that `mamba` is a Conda Forge package, and Conda Forge is standalone and makes no attempt at ensure compatibility with ContinuumIO channels.

If you really want to convert a Miniconda to have Mamba (rather than installing Mambaforge), then you have to source everything from Conda Forge. Here's the steps to convert Miniconda 23.5.2-0 to Mamba 1.5.0 (as demo'd with Docker):

```bash
% docker run --rm -it continuumio/miniconda3:23.5.2-0

## make conda-forge the priority channel
(base) $ conda config --add channels conda-forge

## reinstall everything using only conda-forge
(base) $ conda update --all --override-channels -c conda-forge

## check `conda list` for good measure (omitted)

## install mamba
(base) $ conda install mamba

## verify
(base) $ mamba info

          mamba version : 1.5.0
     active environment : base
    active env location : /opt/conda
            shell level : 1
       user config file : /root/.condarc
 populated config files : /root/.condarc
          conda version : 23.7.3
    conda-build version : not installed
         python version : 3.11.5.final.0
       virtual packages : __archspec=1=x86_64
                          __glibc=2.31=0
                          __linux=5.10.104=0
                          __unix=0=0
       base environment : /opt/conda  (writable)
      conda av data dir : /opt/conda/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /opt/conda/pkgs
                          /root/.conda/pkgs
       envs directories : /opt/conda/envs
                          /root/.conda/envs
               platform : linux-64
             user-agent : conda/23.7.3 requests/2.31.0 CPython/3.11.5 Linux/5.10.104-linuxkit debian/11 glibc/2.31
                UID:GID : 0:0
             netrc file : None
           offline mode : False
```

**Comment by jezdez** [comment]
> It is odd, but in fact the case that even though Conda has embraced libmamba as its backend, the Mamba org no longer recommends using Miniconda base + Mamba (see [Mamba installation docs](https://mamba.readthedocs.io/en/latest/mamba-installation.html#existing-conda-install-not-recommended)).

Not sure what is odd, Mambaforge is a legitimate installer maintained by the community that is now the same as miniforge. The mamba docs don't seem to take that into account yet because we're in a transition at the moment.

> Instead, one should just go straight to Mambaforge (which is https://github.com/conda-forge/conda-forge.github.io/pull/1995).

To be clear, with the advent of conda-libmamba-solver, there was simply no reason for miniforge to not have mamba included, given that in the rollout plan we announced that conda-libmamba-solver would be shipped by default with miniforge and miniconda on our way to make libmamba the default solver for conda in September 2023. Since we already have had to add the various dependencies of conda-libmamba-solver (libmambapy, libmamba, libsolv), it was only a small additional step to also ship the mamba CLI with miniforge.

> The situation is that mamba is a Conda Forge package, and Conda Forge is standalone and makes no attempt at ensuring compatibility with ContinuumIO channels.

Mamba CLI is conda-forge-only (currently), while the Anaconda channel (not ContinuumIO ;) simply only ships the dependencies for conda-libmamba-solver, which are also now part of miniconda.

**Comment by jezdez** [comment]
@gernophil That error message is odd though, where is libmamba 2.0.0 coming from?

```
ImportError: dlopen(~/miniconda3/lib/python3.11/site-packages/libmambapy/bindings.cpython-311-darwin.so, 2): Library not loaded: @rpath/libarchive.13.dylib Referenced from: ~/miniconda3/lib/libmamba.2.0.0.dylib Reason: image not found
```

@jaimergp do you happen to know what's going on here?

**Comment by jaimergp** [comment]
Might the soname in terms of ABI. This is (again) the documented incompatibility between defaults' libarchive and conda-forge libarchive.

See https://github.com/conda-forge/libarchive-feedstock/issues/69 & https://github.com/conda-forge/libarchive-feedstock/pull/70

**Comment by najtin** [comment]
For conda 27.7.3 you can enable the mamba resolver with 
```bash
conda install -n base conda-libmamba-solver
conda config --set solver libmamba
```

**Comment by gernophil** [comment]
> @gernophil That error message is odd though, where is libmamba 2.0.0 coming from?
> 
> ```
> ImportError: dlopen(~/miniconda3/lib/python3.11/site-packages/libmambapy/bindings.cpython-311-darwin.so, 2): Library not loaded: @rpath/libarchive.13.dylib Referenced from: ~/miniconda3/lib/libmamba.2.0.0.dylib Reason: image not found
> ```
> 
> @jaimergp do you happen to know what's going on here?

I installed `mamba` from `conda-forge`. Currently, I cannot reproduce this, since I am back to miniconda 23.1.0. Maybe I find the time to install the latest miniconda in parallel later.

**Comment by najtin** [comment]
> So, if I get this right, there is no real need to use mamba anymore, if I use the conda-libmamba-solver?

This is my understanding of the situation, but i'm by no means a conda expert. I do observe enormous speed ups since i use these settings.

**Comment by jezdez** [comment]
> > So, if I get this right, there is no real need to use mamba anymore, if I use the conda-libmamba-solver?
> 
> This is my understanding of the situation, but i'm by no means a conda expert. I do observe enormous speed ups since i use these settings.

There are other uses for mamba, but with conda-libmamba-solver you get at least the speed of mamba's solver in conda directly. We have some notes on the rollout here: https://conda.org/blog/2023-07-05-conda-libmamba-solver-rollout/

**Comment by gernophil** [comment]
> > > So, if I get this right, there is no real need to use mamba anymore, if I use the conda-libmamba-solver?
> > 
> > 
> > This is my understanding of the situation, but i'm by no means a conda expert. I do observe enormous speed ups since i use these settings.
> 
> There are other uses for mamba, but with conda-libmamba-solver you get at least the speed of mamba's solver in conda directly. We have some notes on the rollout here: https://conda.org/blog/2023-07-05-conda-libmamba-solver-rollout/

Is there already a date for the miniconda September release? Since I would do a fresh miniconda install then. BTW, are you planing on change the policy of the license agreement? With the July release you cannot just press and hold enter to scroll the license agreement since this will lead to declining the license. With this release you have to release enter early enough before you reach the end of the license agreement.

**Comment by gernophil** [comment]
> > > So, if I get this right, there is no real need to use mamba anymore, if I use the conda-libmamba-solver?
> > 
> > 
> > This is my understanding of the situation, but i'm by no means a conda expert. I do observe enormous speed ups since i use these settings.
> 
> There are other uses for mamba, but with conda-libmamba-solver you get at least the speed of mamba's solver in conda directly. We have some notes on the rollout here: https://conda.org/blog/2023-07-05-conda-libmamba-solver-rollout/

btw the command on that page (`conda config --set solver=libmamba`) does not work (`conda config: error: argument --set: expected 2 arguments`). It needs to be `conda config --set solver libmamba` (no `=` sign).

**Comment by jaimergp** [comment]
> And how do I check, if it is libmamba? It does not look like mamba, but like the classic solver.

`libmamba`'s output follows the `conda` UI so it's expected to be closer to classic than mamba's. That said, there are a few signs you can look for:

- If correctly configured, the `user agent` field in `conda info` will mention `solver/libmamba`
- The "Collecting metadata..." step shows more info (channels, platform) than classic
- You'll never see `current_repodata.json` being mentioned.
- The performance differences should be noticeable, hopefully :)

**Comment by jaimergp** [comment]
> With the July release you cannot just press and hold enter to scroll the license agreement since this will lead to declining the license. With this release you have to release enter early enough before you reach the end of the license agreement.

This sounds like a regression in `constructor` 🤔 If you have the time, could you check if the same happens in [Miniforge](https://github.com/conda-forge/miniforge/releases/)? 🙏

**Comment by gernophil** [comment]
> > And how do I check, if it is libmamba? It does not look like mamba, but like the classic solver.
> 
> `libmamba`'s output follows the `conda` UI so it's expected to be closer to classic than mamba's. That said, there are a few signs you can look for:
> 
>     * If correctly configured, the `user agent` field in `conda info` will mention `solver/libmamba`
> 
>     * The "Collecting metadata..." step shows more info (channels, platform) than classic
> 
>     * You'll never see `current_repodata.json` being mentioned.
> 
>     * The performance differences should be noticeable, hopefully :)

So, my ouput of `% conda config --show solver` is `solver: libmamba` and 'conda list | grep libmamba` shows:
```
conda-libmamba-solver     23.5.0          py311hca03da5_0  
libmamba                  1.4.1                h1c5506f_1  
libmambapy                1.4.1           py311h1c5506f_1  
```
However, `conda create -n test python` shows the repodata.json:
```
Channels:
 - conda-forge
 - bioconda
 - defaults
Platform: osx-arm64
Collecting package metadata (repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /Users/gernophil/miniconda3/envs/test

  added / updated specs:
    - python


The following NEW packages will be INSTALLED:

  bzip2              conda-forge/osx-arm64::bzip2-1.0.8-h3422bc3_4 
  ca-certificates    conda-forge/osx-arm64::ca-certificates-2023.7.22-hf0a4a13_0 
  libexpat           conda-forge/osx-arm64::libexpat-2.5.0-hb7217d7_1 
  libffi             conda-forge/osx-arm64::libffi-3.4.2-h3422bc3_5 
  libsqlite          conda-forge/osx-arm64::libsqlite-3.43.0-hb31c410_0 
  libzlib            conda-forge/osx-arm64::libzlib-1.2.13-h53f4e23_5 
  ncurses            conda-forge/osx-arm64::ncurses-6.4-h7ea286d_0 
  openssl            conda-forge/osx-arm64::openssl-3.1.2-h53f4e23_0 
  pip                conda-forge/noarch::pip-23.2.1-pyhd8ed1ab_0 
  python             conda-forge/osx-arm64::python-3.11.5-h47c9636_0_cpython 
  readline           conda-forge/osx-arm64::readline-8.2-h92ec313_1 
  setuptools         conda-forge/noarch::setuptools-68.2.2-pyhd8ed1ab_0 
  tk                 conda-forge/osx-arm64::tk-8.6.12-he1e0b03_0 
  tzdata             conda-forge/noarch::tzdata-2023c-h71feb2d_0 
  wheel              conda-forge/noarch::wheel-0.41.2-pyhd8ed1ab_0 
  xz                 conda-forge/osx-arm64::xz-5.2.6-h57fd34a_0 


Proceed ([y]/n)? n


CondaSystemExit: Exiting.
```
So, this is still the classic solver?

**Comment by gernophil** [comment]
> > With the July release you cannot just press and hold enter to scroll the license agreement since this will lead to declining the license. With this release you have to release enter early enough before you reach the end of the license agreement.
> 
> This sounds like a regression in `constructor` 🤔 If you have the time, could you check if the same happens in [Miniforge](https://github.com/conda-forge/miniforge/releases/)? 🙏

I'll check it and com back to you :).

**Comment by jezdez** [comment]
Lowering the severity because there is a workaround (making sure the packages are all from the same channel) and closing as wontfix since this is being tracked in https://github.com/conda-forge/libarchive-feedstock/issues/69, where the main information how to resolve this is saved.
